File: exs/mundo_1/java/032.java
/*
    ### 032 - Ano bissexto
    Faça um programa que leia um ano qualquer e mostre se ele é BISSEXTO
 */


import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        int ano;
        Scanner scanner = new Scanner(System.in);

        System.out.println("Digite o ano: ");
        ano = scanner.nextInt();

        if (ano % 100 != 0 && ano % 4 == 0 || ano % 400 == 0) {
            System.out.println("É bissexto");
        }
        else
        {
            System.out.println("Não é bissexto");
        }

    }
}

File: exs/mundo_1/rust/032.rs
/*
Desafio 032

Problema: Faça um programa que leia um ano qualquer e mostre se ele é bissexto.
*/

fn main() {
    let mut ano = String::new();

    println!("Insira um ano");
    std::io::stdin().read_line(&mut ano).unwrap();

    let ano = ano
        .trim()
        .parse::<i32>()
        .expect("Valor inserido não é um número válido.");

    if (ano % 4 == 0) && ((ano % 100 != 0) || (ano % 400 == 0)) {
        return println!("O ano de {ano} é bissexto");
    }

    println!("O ano de {ano} não é bissexto")
}
